code/modules/reagents/chem_splash.dm

```diff
@@ -72,6 +72,7 @@
 			var/fraction = 0.5/(2 ** distance) //50/25/12/6... for a 200u splash, 25/12/6/3... for a 100u, 12/6/3/1 for a 50u
 			splash_holder.reaction(A, TOUCH, fraction)
 
+	spawn(0)
```

Comment:
This literally does nothing.

Reply:
Then how does it fix the smoke bombs?

Reply:
I don't actually know how spawn(0) works, but I narrowed it down to the cause as the chems disappear on line 217 in code\modules\reagents\chemistry\recipes\pyrotechnics.dm
They are there before, but not after, so I guessed that it causes it to wait a frame, and it seems to work after a couple of tests.